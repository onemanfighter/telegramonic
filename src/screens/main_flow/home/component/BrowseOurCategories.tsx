import { Box, Divider, Flex, Grid, Text } from '@chakra-ui/react';
import { categories } from './constants';
import { useTranslation } from 'react-i18next';

const BrowseOurCategories = () => {
  const { t } = useTranslation();
  return (
    <Box width={'100%'} p={5}>
      <Box px={8} mb={8}>
        <Divider />
        <Text fontSize={'x-large'} fontWeight={'bold'} colorScheme="black">
          {t('MainHomeScreen.browseOurCategories')}
        </Text>
        <Divider />
      </Box>
      <Flex flexWrap={'wrap'} justifyContent={'space-evenly'}>
        {categories.map((category) => (
          <Box
            key={category.name}
            m={5}
            px={2}
            height={'200'}
            width={'200px'}
            background={require('@assets/images/LogoNoBackground.png')}
            boxShadow={'md'}
            borderRadius={10}
            display={'flex'}
            border={'1px solid black'}
            justifyContent={'center'}
            alignItems={'center'}
          >
            <Text
              fontSize={'md'}
              fontWeight={'bold'}
              textAlign={'center'}
              colorScheme="black"
            >
              {category.name}
            </Text>
          </Box>
        ))}
      </Flex>
    </Box>
  );
};

export default BrowseOurCategories;
