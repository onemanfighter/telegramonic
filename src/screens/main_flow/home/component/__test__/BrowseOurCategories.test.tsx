import { render } from '@testing-library/react';
import BrowseOurCategories from '../BrowseOurCategories';

describe('BrowseOurCategories', () => {
  it('should render correctly', () => {
    const { container } = render(<BrowseOurCategories />);

    expect(container).toMatchSnapshot();
  });
});
