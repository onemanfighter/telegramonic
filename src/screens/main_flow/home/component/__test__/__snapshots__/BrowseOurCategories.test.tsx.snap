// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrowseOurCategories should render correctly 1`] = `
<div>
  <div
    class="css-13v0pi9"
  >
    <div
      class="css-ukzy1g"
    >
      <hr
        aria-orientation="horizontal"
        class="chakra-divider css-1upb9tn"
      />
      <p
        class="chakra-text css-8b95vw"
      >
        Browse our categories
      </p>
      <hr
        aria-orientation="horizontal"
        class="chakra-divider css-1upb9tn"
      />
    </div>
    <div
      class="css-ttzt9k"
    >
      <div
        class="css-1tlv8gh"
      >
        <p
          class="chakra-text css-1ey7oos"
        >
          Best Channels
        </p>
      </div>
      <div
        class="css-1tlv8gh"
      >
        <p
          class="chakra-text css-1ey7oos"
        >
          Best Groups
        </p>
      </div>
      <div
        class="css-1tlv8gh"
      >
        <p
          class="chakra-text css-1ey7oos"
        >
          Best Bots
        </p>
      </div>
      <div
        class="css-1tlv8gh"
      >
        <p
          class="chakra-text css-1ey7oos"
        >
          Best Stickers
        </p>
      </div>
      <div
        class="css-1tlv8gh"
      >
        <p
          class="chakra-text css-1ey7oos"
        >
          Best classic stickers
        </p>
      </div>
      <div
        class="css-1tlv8gh"
      >
        <p
          class="chakra-text css-1ey7oos"
        >
          Best animated stickers
        </p>
      </div>
      <div
        class="css-1tlv8gh"
      >
        <p
          class="chakra-text css-1ey7oos"
        >
          Best stickers packs
        </p>
      </div>
      <div
        class="css-1tlv8gh"
      >
        <p
          class="chakra-text css-1ey7oos"
        >
          Best themes
        </p>
      </div>
      <div
        class="css-1tlv8gh"
      >
        <p
          class="chakra-text css-1ey7oos"
        >
          Games
        </p>
      </div>
    </div>
  </div>
</div>
`;
